ape_safe/_cli.py

```diff
@@ -59,61 +59,96 @@ def _list(cli_ctx, network):
 @ape_cli_context()
 @existing_alias_argument()
 def remove(cli_ctx, alias):
-    safe_container = accounts.containers["safe"]
+    safe_container = cli_ctx.account_manager.containers["safe"]
 
     if alias not in safe_container.aliases:
-        raise
+        raise BadArgumentUsage(f"There is no safe with the alias `{alias}`.")
 
     address = safe_container.load_account(alias).address
     if click.confirm(f"Remove safe {address} ({alias})"):
         safe_container.delete_account(alias)
 
+    cli_ctx.logger.success(f"Safe '{address}' ({alias}) removed.")
+
+
+def _execute_callback(ctx, param, val):
+    if isinstance(val, str):
+        if val in ctx.obj.account_manager.aliases:
+            return ctx.obj.account_manager.load(val)
+        elif val in ctx.obj.account_manager:
+            return ctx.obj.account_manager[val]
+        else:
+            raise BadOptionUsage(
+                "--execute", f"`--execute` value '{val}` not found in local accounts."
+            )
+
+    # Additional handling may occur in command definition.
+    return val
+
 
 @cli.command(
     cls=NetworkBoundCommand, short_help="See pending transactions for a locally-tracked Safe"
 )
+@ape_cli_context()
 @network_option()
+@click.option("sign_with_local_signers", "--sign", is_flag=True)
+@click.option("--execute", is_flag=True, flag_value=True, default=False, callback=_execute_callback)
 @existing_alias_argument(account_type=SafeAccount)
-def pending(network, alias):
-    safe = accounts.load(alias)
-    local_signers = set(signer for signer in accounts if signer.address in safe.signers)
-    if local_signers:
-        click.echo("Local Signer(s) detected!")
-        sign_with_local_signers = click.confirm("Do you want to sign unconfirmed transactions")
-
-    else:
-        sign_with_local_signers = False
-
-    for txn in safe.client.get_transactions(
-        starting_nonce=safe.next_nonce,
-        filter_by_missing_signers=local_signers if sign_with_local_signers else None,
-    ):
-        click.echo(f"Txn {txn.nonce}: ({len(txn.confirmations)}/{txn.confirmationsRequired})")
+def pending(cli_ctx, network, sign_with_local_signers, execute, alias):
+    _ = network  # Needed for NetworkBoundCommand
+    safe = cli_ctx.account_manager.load(alias)
+    submitter = execute
+    if submitter is True:
+        if not safe.local_signers:
+            cli_ctx.abort("Cannot execute without a local signer.")
+
+        submitter = safe.local_signers[0]
+
+    for safe_tx, confirmations in safe.pending_transactions():
+        click.echo(
+            f"Transaction {safe_tx.nonce}: ({len(confirmations)}/{safe.confirmations_required})"
+        )
+
+        # Add signatures, if was requested to do so.
+        if sign_with_local_signers and len(confirmations) < safe.confirmations_required:
+            safe.add_signatures(safe_tx, confirmations)
+            cli_ctx.logger.success(f"Signature added to 'Transaction {safe_tx.nonce}'.")
+
+        if not execute:
```

Comment:
is this backwards? or is it meant to be "add `--execute` or confirm manually to send"?

Reply:
I think you're right - should be execute or confirm

Reply:
In which case, I can see why `--execute` should have 3 states and not be a flag... `None` meaning prompt, otherwise to skip ever being prompted, you specify it directly `--execute true` or `--execute false`. I think that may be how it was but I changed it mistakenly.

Reply:
also looking at the callback, it is meant to take an account

Reply:
done!